src/cookieConsent.js

```diff
@@ -0,0 +1,190 @@
+        padding: theme.spacing(2),
+    },
+    snackbar: isMobile ? ({
+        "& .MuiSnackbarContent-action": {
+            paddingLeft: 0,
+        },
+    }) : ({}),
+}));
+
+const CookiesUsageDialog = ({ open, handleAccept, handleReject, handleToggle }) => {
+    const classes = useStyles()();
+    const isMobile = useMobile();
+
+    return (
+        <Dialog
+            open={open}
+            onClose={handleToggle}
+            classes={{ paper: classes.dialog }}
+        >
+            <DialogTitle>
+                Why do we use cookies?
+            </DialogTitle>
+            <DialogContent style={{ textAlign: "justify" }}>
+                <p>
+                    {"NIJobs uses cookies to collect usage to improve our product."}
+                </p>
+                <p>
+                    {"Regarding privacy issues, we simply collect search parameters and other \
+                        useful metrics without linking them to your identity."}
+                </p>
+                <p>
+                    {"For more information, consult our "}
+                    <Link component={RouterLink} to="/privacy-policy" onClick={handleToggle}>
+                        {"Privacy Policy"}
+                    </Link>
+                    {"!"}
+                </p>
+                <p>
+                    {"Thank you!"}
+                </p>
+            </DialogContent>
+            <DialogActions className={classes.buttonsArea}>
+                <Grid container spacing={2} justifyContent={!isMobile ? "flex-end" : "center"}>
+                    <Grid item xs={12} sm="auto" marginRight={2}>
+                        <Button onClick={handleReject}>Use only essential cookies</Button>
+                    </Grid>
+                    <Grid item xs={12} sm="auto">
+                        <Button color="primary" variant="contained" onClick={handleAccept}>Accept</Button>
+                    </Grid>
+                </Grid>
+            </DialogActions>
+        </Dialog>
+    );
+};
+
+export const CookieConsent = () => {
+    const [open, toggleOpen, resetConsentOpeningState] = useToggle(false);
+
+    const [showWhyModal, toggleWhyModal, setClosedWhyModal] = useToggle(false);
+
+    const handle = (accepted) => {
+        if (accepted) {
+            initAnalytics();
+        } else {
+            clearAnalytics();
+        }
+    };
+
+    useEffect(() => {
+        const cookies = localStorage.getItem("non-essential-cookies-enablement");
+        if (!cookies) {
+            toggleOpen();
+            return;
+        }
+
+        const expired = Date.now() > cookies.expires;
+        if (!expired) {
+            handle(cookies.accepted);
+        } else {
+            toggleOpen();
+        }
+    }, [toggleOpen]);
+
+    const handleAccept = () => {
+        const expires = Date.now() + (COOKIE_ACCEPT_LIFETIME_MONTHS * MONTH_IN_MS);
+        localStorage.setItem("non-essential-cookies-enablement", {
```

Comment:
`setItem` allows only strings both in key and value [0]. This means that the value is currently being converted to string (JS automatically calls `.toString` on the object you pass as value, resulting in `[object Object]` as you can confirm by inspecting the localStorage after accepting cookies.

You need to `JSON.stringify` the object when setting, and `JSON.parse` when getting the values (in every get and set you have here, respectively).

---

[0] https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem